### PR TITLE
Refactor/add sqlalchemy

### PR DESCRIPTION
**Pull request by timbeccue**
This PR corresponds to [this](https://www.pivotaltracker.com/story/show/174915116) story. The goal was to use sqlalchemy for all database connections. I added tests along the way too, since they were mostly non-existent. 

All the endpoint methods that connect to postgres have been moved into `api/db.py`. 
Corresponding tests are in `api/tests/unit/test_db.py`.
Previously, most of these methods sat in `api/handler.py`. 

All commits besides the first one were iterative attempts at getting travis to use aws credentials.

### CODE REVIEW

File: api/db.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    Args: 
+        db_address (str): address for the database to query
+        base_filename (str): the filename (without the ex-version or .extension)
+                for identifying the header we want. 
+
+    Returns:
+        dict: the image package representing the image with common metadata.
+
+    """
+    query_filters = [
```
Comment (mgdaily) [inline api/db.py:232]: Doesn't look like this list of criteria is actually being used in your query, perhaps a mistake?

File: api/db.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+            images = session.query(Image)\
+                .order_by(Image.sort_date.desc())\
+                .filter(*query_filters)\
+                .limit(100)\
+                .all()
+            session.expunge_all()
+        except Exception as e:
+            logger.exception("Error in filtered images query.")
+            raise
+    image_pkgs = [i.get_image_pkg() for i in images]
```
Comment (mgdaily) [inline api/db.py:214]: I can hear Curtis in my head: _Good variable names, young Padawan_

`[image.get_image_pkg() for image in images]` ?

File: api/db.py (hunk 1 above)
Comment (timbeccue) [reply api/db.py:232]: good catch! fixed.

File: api/db.py [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+    with get_session(db_address=db_address) as session:
+        images = session.query(Image)\
+            .order_by(Image.sort_date.desc())\
+            .filter(*query_filters)\
+            .limit(number_of_images)\
+            .all()
+        session.expunge_all()
+    image_pkgs = [i.get_image_pkg() for i in images]
+    print(json.dumps(image_pkgs, indent=2))
```
[the remark this answers is not in the document]
Comment (timbeccue) [reply api/db.py:158]: another good catch, thanks

File: api/db.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    Returns:
+        dict: the image package representing the image with common metadata.
+
+    """
+    query_filters = [
+        Image.base_filename == base_filename, # match filenames
+        Image.ex10_jpg_exists.is_(True)       # the jpg must exist
+    ]
+    with get_session(db_address=db_address) as session:
+        try:
```
[the remark this answers is not in the document]
Comment (timbeccue) [reply api/db.py:237]: Oh yeah, that makes sense. I think adding the log messages in the handlers is a good way to go.

File: api/db.py (hunk 2 above)
Comment (timbeccue) [reply api/db.py:214]: Haha fair enough!

**Review by mgdaily — APPROVED** [review]

**Pull request merged**